qiskit/synthesis/linear/cz_depth_lnn.py

```diff
@@ -0,0 +1,183 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Optimize the synthesis of an n-qubit circuit contains only CZ gates for
+linear nearest neighbor (LNN) connectivity, using CX and phase (S, Sdg or Z) gates.
+The two-qubit depth of the circuit is bounded by 2*n+2.
+
+References:
+    [1]: Dmitri Maslov, Martin Roetteler,
+         Shorter stabilizer circuits via Bruhat decomposition and quantum circuit transformations,
+         `arXiv:1705.09176 <https://arxiv.org/abs/1705.09176>`_.
+"""
+
+import numpy as np
+from qiskit.circuit import QuantumCircuit
+
+
+def _append_cx_stage1(qc, n):
+    """A single layer of CX gates."""
+    for i in range(int(n / 2)):
+        qc.cx(2 * i, 2 * i + 1)
+    for i in range(int((n + 1) / 2) - 1):
+        qc.cx(2 * i + 2, 2 * i + 1)
+    return qc
+
+
+def _append_cx_stage2(qc, n):
+    """A single layer of CX gates."""
+    for i in range(int(n / 2)):
+        qc.cx(2 * i + 1, 2 * i)
+    for i in range(int((n + 1) / 2) - 1):
+        qc.cx(2 * i + 1, 2 * i + 2)
+    return qc
+
+
+def _append_cx_stage(qc, n):
+    """Append a depth 2 layer of CX gates"""
+    qc.compose(_append_cx_stage1(qc, n))
```

Comment:
Is there a reason to use `compose` method and not `append` method here?

Reply:
Yes, append raises an error, and compose doesn't